selectrum.el

```diff
@@ -799,39 +832,60 @@ PRED defaults to `minibuffer-completion-predicate'."
       (setq deactivate-mark nil))
     (setq-local selectrum--init-p nil)))
 
-(defun selectrum--first-lines (candidates)
+(defun selectrum--ensure-single-lines (candidates)
   "Return list of single line CANDIDATES.
-Multiline canidates are merged into a single line."
-  (let ((onelines ()))
-    (dolist (cand candidates (nreverse onelines))
+Multi-line canidates are merged into a single line."
+  (let* ((single-line-candidates ())
+
+         ;; The indicators are the same for all multi-line candidates, and so
+         ;; only need to be gotten from `selectrum-candidate-transformations'
+         ;; once.
+         (match-transformation
+          (alist-get 'match selectrum-candidate-transformations))
+         (match-display (plist-get match-transformation :indicator))
+         (match-face (plist-get match-transformation :face))
+         (truncation-transformation
+          (alist-get 'truncation selectrum-candidate-transformations))
+         (truncation-display (plist-get truncation-transformation :indicator))
+         (truncation-face (plist-get truncation-transformation :face))
+         (newline-transformation
+          (alist-get 'newline selectrum-candidate-transformations))
+         (newline-display (plist-get newline-transformation :indicator))
+         (newline-face (plist-get newline-transformation :face))
+         (whitespace-transformation
+          (alist-get 'whitespace selectrum-candidate-transformations))
+         (whitespace-display (plist-get whitespace-transformation :indicator))
+         (whitespace-face (plist-get whitespace-transformation :face)))
```

Comment:
There is a caveat when let binding symbols with dashes in their name: It may happen that you accidentally use name name which is a declared as a special variable in other packages.